### ISSUE TITLE
delta.analyze fails with LinAlgError for small sample sizes

### ISSUE THREAD

**Issue by willu47** [SALib/SALib#152]
See #151 for discussion on this error.

Use the test file below to reproduce error:

```python
from SALib.analyze.delta import analyze
import numpy as np

def test_bug_delta_linalg():

    problem = problem = {
                'num_vars': 3,
                'names': ['K_IO_AOBden', 'K_SO_AOBden', 'K_SNO_AOBden'],
                'bounds': [[0.01, 11], [0.01, 13.2], [0.08, 4.3]]
                         }

    X = np.array([[2.12591411, 1.8200793, 0.10082878],
                  [6.21754537, 5.85214066, 2.8846046],
                  [8.01209459, 12.95017617, 1.00034018],
                  [10.72587215, 9.03629573, 1.88932555],
                  [4.17005179, 1.30648717, 3.7287254],
                  [3.24299291, 3.44273292, 4.01808892],
                  [8.80676551, 10.03481953, 1.43068963],
                  [0.29322542, 6.80514303, 2.43057428]])

    Y = np.arange(8)

    analyze(problem, X, Y, num_resamples=100)
```

**Comment by ConnectedSystems** [comment]
Based on discussion here, suggest testing for singular matrix and throwing error indicating an inadequate sample size:

```python
from __future__ import division

from SALib.analyze.delta import analyze
import numpy as np
import sys


def test_bug_delta_linalg():

    problem = problem = {
                'num_vars': 3,
                'names': ['K_IO_AOBden', 'K_SO_AOBden', 'K_SNO_AOBden'],
                'bounds': [[0.01, 11], [0.01, 13.2], [0.08, 4.3]]
                         }

    X = np.array([[2.12591411, .8200793, 0.10082878],
                  [6.21754537, 5.85214066, 2.8846046],
                  [8.01209459, 12.95017617, 1.00034018],
                  [10.72587215, 9.03629573, 1.88932555],
                  [4.17005179, 1.30648717, 3.7287254],
                  [3.24299291, 3.44273292, 4.01808892],
                  [8.80676551, 10.03481953, 1.43068963],
                  [0.29322542, 6.80514303, 2.43057428]])
    
    # This would go into `analyze()`
    if np.linalg.cond(X) < 1/np.finfo(X.dtype).eps:
        raise ValueError("Singular matrix detected - sample size may be too small")
    # End if

    Y = np.arange(8)
    
    analyze(problem, X, Y, num_resamples=100)
```

**Comment by willu47** [comment]
Thanks @ConnectedSystems - this would be a neat fix - do you want to submit a PR for this?  One suggestion would be to relate the error message to the input parameters e.g. sample size `n` is too small.

**Comment by willu47** [comment]
Closed by #225
